File: src/lecture20250319/Calculator.java
package lecture20250319;

public class Calculator {
    public static void main(String[] args) {
        java.util.Scanner scanner = new java.util.Scanner(System.in);
        char operator;
        int number1, number2;
        double result;
        
        System.out.println("Welcome to Simple Calculator!");
        
        while(true) {
            System.out.println("First input an operator (+, -*./, ^) ...then two integer numbers and i will calculate the result");
            System.out.println("An empty operator terminates the program!");
            
            System.out.print("Operator: ");
            String input = scanner.nextLine();
            
            // Prüfe auf leere Eingabe
            if (input.isEmpty() || input.trim().isEmpty()) {
                System.out.println("Program terminated.");
                break;
            }
            
            operator = input.charAt(0);
            
            System.out.print("Number 1:");
            number1 = scanner.nextInt();
            
            System.out.print("Number 2:");
            number2 = scanner.nextInt();
            
            // Verbrauche das verbleibende Zeilenende
            scanner.nextLine();
            
            if (operator == '+') {
                result = number1 + number2;
            } else if (operator == '-') {
                result = number1 - number2;
            } else if (operator == '*') {
                result = number1 * number2;
            } else if (operator == '/') {
                result = (double)number1 / number2;
            } else if (operator == '^') {
                result = Math.pow(number1, number2);
            } else {
                System.out.println("Invalid operator");
                continue;
            }
            
            System.out.printf("%d %c %d = %.2f\n", number1, operator, number2, result);
        }
        
        scanner.close();
    }
}

File: calculator.c
#include <stdio.h>
#include <math.h>

int main() {
    char operator;
    int number1, number2;
    float result;
    
    printf("Welcome to Simple Calculator!\n");

    while(1) {
        printf("First input an operator (+, -*./, ^) ...then two integer numbers and i will calculate the result\n");
        printf("An empty operator terminates the program!\n");
        
        printf("Operator (+, -, *, /, ^, x for exit): ");
        scanf(" %c", &operator);
        
        // Hier prüfen wir auf einen speziellen Wert wie 'q' oder 'x' zum Beenden
        if (operator == 'q' || operator == 'x') {
            printf("Program terminated.\n");
            break;
        }
        
        printf("Number 1:");
        scanf("%d", &number1);
        
        printf("Number 2:");
        scanf("%d", &number2);
        
        if (operator == '+') {
            result = number1 + number2;
        } else if (operator == '-') {
            result = number1 - number2;
        } else if (operator == '*') {
            result = number1 * number2;
        } else if (operator == '/') {
            result = (float)number1 / number2;
        } else if (operator == '^') {
            result = pow(number1, number2);
        } else {
            printf("Invalid operator\n");
            // Eingabepuffer leeren
            while (getchar() != '\n');
            continue;
        }
        
        printf("%d %c %d = %.2f\n", number1, operator, number2, result);
        
        // Eingabepuffer leeren
        while (getchar() != '\n');
    }
    
    return 0;
}